### PR TITLE
Fix calculation of pixel diagonal half-length - qgsalgorithmlinedensity.cpp

### PR DESCRIPTION
**Pull request by havatv**
## Description

The calculation of the length of the pixel diagonal was wrong, so I fixed it.

### CODE REVIEW

**Comment by havatv** [comment]
Travis complains about indentation, but my PR keeps the original indentation. So what should I do?

**Comment by roya0045** [comment]
> Travis complains about indentation, but my PR keeps the original indentation. So what should I do?

I would think that standardizing to the current style would be better, as its simply adding a space and will prevent any future complains from the test.

**Comment by havatv** [comment]
Thanks, @roya0045!
Sorry about my ignorance, but where do I add the space?

**Comment by roya0045** [comment]
See the test results https://travis-ci.org/qgis/QGIS/jobs/651910894?utm_medium=notification&utm_source=github_status

**Comment by havatv** [comment]
I did checked the test results:

```
File src/analysis/processing/qgsalgorithmlinedensity.cpp needs indentation
Required indentation updates:
--- src/analysis/processing/qgsalgorithmlinedensity.cpp.prepare	2020-02-18 17:46:36.264417121 +0600
+++ src/analysis/processing/qgsalgorithmlinedensity.cpp	2020-02-18 17:40:24.000000000 +0600
@@ -84,7 +84,7 @@
   mPixelSize = parameterAsDouble( parameters, QStringLiteral( "PIXEL_SIZE" ), context );
   mSearchRadius = parameterAsDouble( parameters, QStringLiteral( "RADIUS" ), context );
-  if ( mSearchRadius < 0.5 * std::sqrt( 2 * std::pow( mPixelSize, 2) ) )
+  if ( mSearchRadius < 0.5 * std::sqrt( 2 * std::pow( mPixelSize, 2 ) ) )
     throw QgsProcessingException( QObject::tr( "Raster cells must be fully contained by the search circle. Therefore, "
                                   "the search radius must not be smaller than half of the pixel diagonal." ) );
```

But did not understand where "simply adding a space" should be done.

**Comment by roya0045** [comment]
Directly between 2 and ).

File: src/analysis/processing/qgsalgorithmlinedensity.cpp [hunk 1]
```diff
@@ -84,9 +84,9 @@ bool QgsLineDensityAlgorithm::prepareAlgorithm( const QVariantMap &parameters, Q
   mPixelSize = parameterAsDouble( parameters, QStringLiteral( "PIXEL_SIZE" ), context );
 
   mSearchRadius = parameterAsDouble( parameters, QStringLiteral( "RADIUS" ), context );
-  if ( mSearchRadius < std::sqrt( mPixelSize ) / 2 )
+  if ( mSearchRadius < 0.5 * std::sqrt( 2 * std::pow( mPixelSize, 2) ) )
```
Comment (havatv) [inline src/analysis/processing/qgsalgorithmlinedensity.cpp]:
```suggestion
  if ( mSearchRadius < 0.5 * std::sqrt( 2 * std::pow( mPixelSize, 2 ) ) )
```

**Comment by havatv** [comment]
Ah - it's a diff that provides the solution! :-)
I was confused that Travis uses the phrase "needs indentation"...

**Comment by roya0045** [comment]
Yeah its a spacing/indentation test. Name is not optimal but once you know its fine.

**Comment by nyalldawson** [comment]
@root676 can you take a look at this?

**Review by root676 — COMMENTED** [review]
Thank you for fixing this - I think I mixed up some formulas there.

File: src/analysis/processing/qgsalgorithmlinedensity.cpp [hunk 2]
```diff
@@ -84,9 +84,9 @@ bool QgsLineDensityAlgorithm::prepareAlgorithm( const QVariantMap &parameters, Q
   mPixelSize = parameterAsDouble( parameters, QStringLiteral( "PIXEL_SIZE" ), context );
 
   mSearchRadius = parameterAsDouble( parameters, QStringLiteral( "RADIUS" ), context );
-  if ( mSearchRadius < std::sqrt( mPixelSize ) / 2 )
+  if ( mSearchRadius < 0.5 * std::sqrt( 2 * std::pow( mPixelSize, 2 ) ) )
```
Comment (havatv) [inline src/analysis/processing/qgsalgorithmlinedensity.cpp]:
```suggestion
  if ( mSearchRadius < 0.5 * mPixelSize * std::sqrt( 2 ) )  
```

**Review by root676 — APPROVED** [review]

File: src/analysis/processing/qgsalgorithmlinedensity.cpp [hunk 3]
```diff
@@ -84,9 +84,9 @@ bool QgsLineDensityAlgorithm::prepareAlgorithm( const QVariantMap &parameters, Q
   mPixelSize = parameterAsDouble( parameters, QStringLiteral( "PIXEL_SIZE" ), context );
 
   mSearchRadius = parameterAsDouble( parameters, QStringLiteral( "RADIUS" ), context );
-  if ( mSearchRadius < std::sqrt( mPixelSize ) / 2 )
+  if ( mSearchRadius < 0.5 * mPixelSize * std::sqrt( 2 ) )
```
Comment (havatv) [inline src/analysis/processing/qgsalgorithmlinedensity.cpp]:
```suggestion
  if ( mSearchRadius < 0.5 * mPixelSize * std::sqrt( 2 ) )
```

**Pull request merged**